shell/mixins/brand.js

```diff
@@ -61,7 +68,23 @@ export default {
     },
 
     cspAdapter() {
-      return findBy(this.apps, 'metadata.name', 'rancher-csp-adapter' );
+      if (!this.canCalcCspAdapter) {
+        // We only have a watch on cspAdapter to kick off persisting the brand setting.
+        // So we need to ensure we don't return an undefined here... which would match the undefined gave if no csp app was found...
+        // .. and wouldn't kick off the watcher
+        return '';
+      }
+
+      // Note! these used to be `findBy(this.app)` however for that case we lost reactivity on the collection
+      // (computed fires before fetch, fetch happens and update apps, computed would not fire again - even with vue.set)
+      // So use `.find` here instead
+      return this.apps.find((a) => a.metadata.name === 'rancher-csp-adapter');
+    },
+
+    canCalcCspAdapter() {
+      // We need to take consider the loggedIn state, as the brand mixin is used in the logout page where we can be in a mixed state
+      // (things in store but user has no auth to make changes)
+      return this.loggedIn && this.haveAppsAndSettings;
```

Comment:
this block is also different from https://github.com/rancher/dashboard/pull/9092/files#diff-556137f68d06f23031944f124827daa23f6f1cb5c7c6fe4ade07260010874f9eR70-R80

Reply:
As per https://github.com/rancher/dashboard/pull/9269#discussion_r1261050249